packages/vite/src/node/server/moduleGraph.ts

```diff
@@ -179,20 +182,33 @@ export class ModuleGraph {
     return noLongerImported
   }
 
-  async ensureEntryFromUrl(
-    rawUrl: string,
-    ssr?: boolean,
-    setIsSelfAccepting = true
-  ): Promise<ModuleNode> {
-    const [url, resolvedId, meta] = await this.resolveUrl(rawUrl, ssr)
-    let mod = this.idToModuleMap.get(resolvedId)
+  ensureEntryFromResolved(
+    [url, resolvedId, meta]: ResolvedUrl,
+    isSelfAccepting?: boolean
+  ): ModuleNode {
+    const unhashedId = removeDepVersionQuery(resolvedId)
+    if (this.idToModuleMap.has(unhashedId)) {
+      resolvedId = unhashedId
```

Comment:
We have to strip the version query here and see if a module node exists without, as we can't safely remove it from inside `ensureEntryFromUrl`.